src/polymatheia/data/reader.py

```diff
@@ -322,3 +323,56 @@ def __iter__(self):
         if self._file.seekable():
             self._file.seek(0)
         return NavigableDictIterator(iter(DictReader(self._file)))
+
+
+class SRUExplainRecordReader(object):
+    """The class:`~polymatheia.data.reader.SRUExplainRecordReader` is a container for SRU Explain Records."""
+
+    def __init__(self, url):
+        """Construct a new class:`~polymatheia.data.reader.SRUExplainRecordReader`.
+
+        :param url: The base URL of the SRU server
+        :type url: ``str``
+        """
+        self._url = url
+        self._explain = SRUpy(self._url).explain()
+        self.data = NavigableDict(self._explain)
+        self.schemas = [(schema["@name"], schema.title)
+                        for schema in self.data.explain.schemaInfo.schema]
+        self.echo = NavigableDict(self._explain.echo)
+
+
+class SRURecordReader(object):
+    """The :class:`~polymatheia.data.reader.SRURecordReader` is an iteration container for Records fetched via SRU.
+
+    The underlying library (SRUpy) automatically handles the continuation parameters, allowing for simple iteration.
+    """
+
+    def __init__(self, url, as_navigable_dict=True, **kwargs):
+        """Construct a new :class:`~polymatheia.data.reader.SRURecordReader`.
+
+        :param url: The base URL of the SRU endpoint
+        :type url: ``str``
+        :param as_navigable_dict: Whether the records will be returned as they come from the SRU server (usually XML)
+        or as an instance of :class:`~polymatheia.data.NavigableDictIterator` (default)
+        :type as_navigable_dict: ``boolean``
+        :param kwargs: Requests parameters that will be sent to the SRU server
+        """
+        self._url = url
+        self._as_navigable_dict = as_navigable_dict
```

Comment:
Please remove the as_navigable_dict option, as the output will not be usable with any other part of polymatheia. If users want the raw XML, they should use the underlying library directly. Polymatheia is a bit opinionated and uses NavigableDicts everywhere :-).